cobbler/modules/authentication/pam.py

```diff
@@ -111,20 +118,26 @@ class PamConv(Structure):
 
 def authenticate(api_handle, username, password):
     """
-    Returns True if the given username and password authenticate for the
-    given service.  Returns False otherwise
+    :param api_handle: Used for resolving the the pam service name and getting the Logger.
+    :param username:The username to log in with.
+    :type username: str
+    :param password: The password to log in with.
+    :type password: str
+    :returns: True if the given username and password authenticate for the given service. Otherwise False
+    :rtype: bool
     """
 
     @CONV_FUNC
     def my_conv(n_messages, messages, p_response, app_data):
-        """Simple conversation function that responds to any
-        prompt where the echo is off with the supplied password"""
+        """
+        Simple conversation function that responds to any prompt where the echo is off with the supplied password
+        """
         # Create an array of n_messages response objects
         addr = CALLOC(n_messages, sizeof(PamResponse))
         p_response[0] = cast(addr, POINTER(PamResponse))
         for i in range(n_messages):
             if messages[i].contents.msg_style == PAM_PROMPT_ECHO_OFF:
-                pw_copy = STRDUP(str(password))
+                pw_copy = STRDUP(password.encode())
```

Comment:
So currently this is realted to #2400 , the code from the pypi package is this: https://github.com/FirefighterBlu3/python-pam/blob/master/pam.py#L157

Thus I think this is reasonable. In a follow up I will replace our code with the package.

Reply:
That's right. If you pass something else, then it would crash here. But on the other hand it would crash in every case. So I guess that's okay for now… unless we want explicit type checking.